retrofit-adapters/rxjava/src/test/java/retrofit2/adapter/rxjava/ObservableTest.java

```diff
@@ -83,6 +84,22 @@
     }
   }
 
+  @Test public void bodyRespectsBackpressure() {
+    server.enqueue(new MockResponse().setBody("Hi"));
+
+    TestSubscriber<String> subscriber = new TestSubscriber<>(0);
+    Observable<String> o = service.body();
+
+    o.subscribe(subscriber);
+    assertThat(server.getRequestCount()).isEqualTo(0);
+
+    subscriber.requestMore(1);
+    assertThat(server.getRequestCount()).isEqualTo(1);
```

Comment:
Requests are synchronous!

On Tue, Mar 29, 2016, 9:46 PM Jesse Wilson <notifications@github.com> wrote:

> In
> retrofit-adapters/rxjava/src/test/java/retrofit2/adapter/rxjava/ObservableTest.java
> <https://github.com/square/retrofit/pull/1702#discussion_r57827518>:
>
> > @@ -83,6 +84,22 @@
> >      }
> >    }
> >
> > +  @Test public void bodyRespectsBackpressure() {
> > +    server.enqueue(new MockResponse().setBody("Hi"));
> > +
> > +    TestSubscriber<String> subscriber = new TestSubscriber<>(0);
> > +    Observable<String> o = service.body();
> > +
> > +    o.subscribe(subscriber);
> > +    assertThat(server.getRequestCount()).isEqualTo(0);
> > +
> > +    subscriber.requestMore(1);
> > +    assertThat(server.getRequestCount()).isEqualTo(1);
>
> looks racy
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/square/retrofit/pull/1702/files/c107778914815466bda00878c4cbc6b799a6e133#r57827518>
>